node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java

```diff
@@ -60,13 +61,16 @@ public Preparer(NodeRepository nodeRepository, FlagSource flagSource, Optional<H
     private List<Node> prepareNodes(ApplicationId application, ClusterSpec cluster, NodeSpec requestedNodes, int wantedGroups) {
         List<Node> surplusNodes = findNodesInRemovableGroups(application, cluster, wantedGroups);
 
-        MutableInteger highestIndex = new MutableInteger(findHighestIndex(application, cluster));
+        NodeList nodesInCluster = nodeRepository.nodes().list(Node.State.allocatedStates().toArray(new Node.State[0]))
```

Comment:
NodeList nodesInCluster = nodeRepository.nodes().list().owner(application).cluster(cluster);

Reply:
`cluster(cluster.id())` in that case. The code I replaced also checked that `type` was the same—I modified this to be in line with what the actual provisioning code does, which is to use `satisfies(...)`.

Reply:
.. but you're saying that's irrelevant, and that id is the only thing that matters (which I was I was expecting, tbh.)

Reply:
Yes (and also that these filters out unallocated).

Reply:
Thanks, good point.